server/sonar-web/src/main/js/apps/quality-profiles/details/ChangeParentForm.js

```diff
@@ -0,0 +1,138 @@
+import { sortBy } from 'lodash';
+import { changeProfileParent } from '../../../api/quality-profiles';
+import { translate } from '../../../helpers/l10n';
+import type { Profile } from '../propTypes';
+
+type Props = {
+  onChange: () => void,
+  onClose: () => void,
+  onRequestFail: Object => void,
+  profile: Profile,
+  profiles: Array<Profile>
+};
+
+type State = {
+  loading: boolean,
+  selected: ?string
+};
+
+export default class ChangeParentForm extends React.PureComponent {
+  mounted: boolean;
+  props: Props;
+  state: State = {
+    loading: false,
+    selected: null
+  };
+
+  componentDidMount() {
+    this.mounted = true;
+  }
+
+  componentWillUnmount() {
+    this.mounted = false;
+  }
+
+  handleCancelClick = (event: Event) => {
+    event.preventDefault();
+    this.props.onClose();
+  };
+
+  handleSelectChange = (option: { value: string }) => {
+    this.setState({ selected: option.value });
+  };
+
+  handleFormSubmit = (event: Event) => {
+    event.preventDefault();
+
+    const parent = this.state.selected;
+
+    if (parent != null) {
+      this.setState({ loading: true });
+      changeProfileParent(this.props.profile.key, parent).then(this.props.onChange).catch(error => {
+        if (this.mounted) {
+          this.setState({ loading: false });
+        }
+        this.props.onRequestFail(error);
+      });
+    }
+  };
+
+  render() {
+    const { profiles } = this.props;
+
+    const options = [
+      { label: translate('none'), value: '' },
+      ...sortBy(profiles, 'name').map(profile => ({ label: profile.name, value: profile.key }))
+    ];
+
+    return (
+      <Modal
+        isOpen={true}
+        contentLabel={translate('quality_profiles.change_parent')}
+        className="modal"
+        overlayClassName="modal-overlay"
+        onRequestClose={this.props.onClose}>
+
+        <form id="change-profile-parent-form" onSubmit={this.handleFormSubmit}>
+          <div className="modal-head">
+            <h2>{translate('quality_profiles.change_parent')}</h2>
+          </div>
+          <div className="modal-body">
+            <div className="modal-field">
+              <label htmlFor="change-profile-parent">
+                {translate('quality_profiles.parent')}: <em className="mandatory">*</em>
+              </label>
+              <Select
+                clearable={false}
+                id="change-profile-parent"
+                name="parentKey"
+                onChange={this.handleSelectChange}
+                options={options}
+                value={
+                  this.state.selected != null
+                    ? this.state.selected
+                    : this.props.profile.parentKey || ''
+                }
+              />
+            </div>
+          </div>
+          <div className="modal-foot">
+            {this.state.loading && <i className="spinner spacer-right" />}
+            <button
+              disabled={this.state.loading || this.state.selected == null}
```

Comment:
Same as rename and copy, we can submit when selecting the same parent as before.
In the end I don't know if we should really prevent the user from submitting, maybe we should just not do the api call.